presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaMetadata.java

```diff
@@ -349,6 +349,9 @@ private boolean isColumnsEnumeratingTable(SchemaTableName schemaTableName)
         Optional<Set<String>> tables = filterString(constraint, TABLE_NAME_COLUMN_HANDLE);
         if (tables.isPresent()) {
             return prefixes.stream()
+                    .flatMap(prefix -> prefix.getSchemaName()
+                            .map(schemaName -> Stream.of(prefix))
+                            .orElseGet(() -> listSchemaNames(session)))
                     .flatMap(prefix -> tables.get().stream()
                             .filter(this::isLowerCase)
```

Comment:
shouldn't we remove this `filter` (or below `map`) if we are going to map to lower case anyway? Can you add a test for that `WHERE table_name = 'NATION'`?